tests/python/pants_test/base/test_payload_field.py

```diff
@@ -120,6 +121,29 @@ def fingerprint(self):
     self.assertEquals(fingerprinted_field1.fingerprint(), fingerprinted_field1_same.fingerprint())
     self.assertNotEquals(fingerprinted_field1.fingerprint(), fingerprinted_field2.fingerprint())
 
+  def test_set_of_primitives_field(self):
+    # Should preserve `None` values.
+    self.assertEqual(PrimitivesSetField(None).value, None)
+
+    def sopf(underlying):
+      return PrimitivesSetField(underlying).fingerprint()
+    self.assertEqual(
+      sopf({'one', 'two'}),
+      sopf({'two', 'one'}),
+    )
+    self.assertEqual(
+      sopf(['one', 'two']),
+      sopf(['two', 'one']),
+    )
+    self.assertEqual(
+      sopf(None),
+      sopf(None),
+    )
+    self.assertNotEqual(
+      sopf(None),
+      sopf(['one']),
+    )
```

Comment:
Could you add an explicit assertion for differentiating "not set" from "empty" to these?
```
self.assertNotEqual(
  sopf(None),
  sopf([]),
)
```

Reply:
Hah, good call.